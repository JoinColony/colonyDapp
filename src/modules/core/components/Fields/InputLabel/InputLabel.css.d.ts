export const main: string;
export const themeFat: string;
export const themeUnderlined: string;
export const themeMinimal: string;
export const directionHorizontal: string;
export const horizontal: string;
export const colorSchemaDark: string;
export const labelText: string;
export const colorSchemaGrey: string;
export const error: string;
export const help: string;
export const helpAlignRight: string;
export const extra: string;
export const stateScreenReaderOnly: string;
export const sizeMedium: string;
