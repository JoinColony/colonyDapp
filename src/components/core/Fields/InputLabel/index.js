/* @flow */

export { default } from './InputLabel.jsx';
