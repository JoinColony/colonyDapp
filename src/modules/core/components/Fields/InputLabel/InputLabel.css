.main {
  display: block;
  margin-bottom: 6px;
  width: 100%;
  position: relative;
  font-size: var(--size-smallish);
  font-weight: var(--weight-bold);
  text-align: left;
  line-height: calc(var(--size-small) + 2px);
}

.themeFat {
  composes: main;
}

.themeUnderlined {
  composes: main;
}

.themeMinimal {
  composes: main;
}

.directionHorizontal {
  display: inline-block;
  margin-bottom: 0;
  margin-right: 6px;
  width: auto;
  vertical-align: super;
}

.colorSchemaDark .labelText {
  color: var(--grey-5);
}

.colorSchemaGrey .labelText {
  font-weight: var(--weight-bold);
  color: var(--dark);
}

.colorSchemaLightGrey .labelText {
  color: var(--temp-grey-blue-7);
}

.error, .help {
  display: inline-block;
  margin-left: 5px;
  font-size: var(--size-tiny);
  font-weight: 400;
  line-height: 1;
}

.error {
  color: var(--danger);
}

.help {
  color: var(--text);
}

.helpAlignRight {
  display: flex;
  justify-content: space-between;
  align-items: baseline;
}

.extra {
  position: absolute;
  top: 0;
  right: 0;
}

.stateScreenReaderOnly {
  margin: -1px;
  padding: 0;
  height: 1px;
  width: 1px;
  position: absolute;
  overflow: hidden;
  border: 0;
  clip: rect(0, 0, 0, 0);
}

.sizeMedium {
  font-size: var(--size-medium-l);
}
