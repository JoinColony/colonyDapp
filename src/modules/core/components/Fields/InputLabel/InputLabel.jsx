/* @flow */

import React from 'react';
import type { IntlShape, MessageDescriptor } from 'react-intl';

import type { Appearance } from '$types/css';
import { getMainClasses } from '$utils/css';

import styles from './InputLabel.css';

const displayName = 'core.InputLabel';

type Props = {
  appearance?: Appearance,
  error?: string,
  help?: string | MessageDescriptor,
  helpValues?: { [string]: string },
  id: string,
  intl: IntlShape,
  label?: string | MessageDescriptor,
  labelValues?: { [string]: string },
}

const InputLabel = ({ appearance, error, help, helpValues, id, intl: { formatMessage }, label: inputLabel, labelValues }: Props) => {
  const helpText = (typeof help == 'object') ? formatMessage(help, helpValues) : help;
  const labelText = (typeof inputLabel == 'object') ? formatMessage(inputLabel, labelValues) : inputLabel;
  return (
    <label
      className={getMainClasses(appearance, styles)}
      id={`${id}_label`}
      htmlFor={id}
    >
      <span className={styles.labelText}>{labelText}</span>
      { error && appearance && appearance.direction !== 'horizontal' ?
        <span className={styles.error}>{error}</span>
        :
        helpText && <span className={styles.help}>({helpText})</span>
      }
    </label>
  );
};

InputLabel.displayName = displayName;

export default InputLabel;
