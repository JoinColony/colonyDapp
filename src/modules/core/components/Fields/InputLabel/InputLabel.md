### Simple input label

```js
<InputLabel label="Label text" />
```

### Input label with help text

```js
<InputLabel label="Label text" help="(With a little help from my friends)" />
```
