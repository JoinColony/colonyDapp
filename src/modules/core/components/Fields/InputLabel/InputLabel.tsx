import React, { ReactNode } from 'react';
import { MessageDescriptor, useIntl } from 'react-intl';

import { SimpleMessageValues } from '~types/index';
import { getMainClasses } from '~utils/css';

import styles from './InputLabel.css';

const displayName = 'InputLabel';

interface Appearance {
  theme?: 'fat' | 'underlined' | 'minimal' | 'dotted';
  direction?: 'horizontal';
  colorSchema?: 'dark' | 'grey' | 'transparent' | 'info';
  helpAlign?: 'right';
  size?: 'small' | 'medium';
}

interface Props {
  /** Appearance object */
  appearance?: Appearance;

  /** Extra node to render on the top right in the label */
  extra?: ReactNode;

  /** Help text (will appear next to label text) */
  help?: string | MessageDescriptor;

  /** Values for help text (react-intl interpolation) */
  helpValues?: SimpleMessageValues;

  /** `id` attribute value of accompanied input field */
  inputId?: string;

  /** Label text */
  label: string | MessageDescriptor;

  /** Values for label text (react-intl interpolation) */
  labelValues?: SimpleMessageValues;

  /** Should only be visible for screenreaders, but not for display users */
  screenReaderOnly?: boolean;
}

const InputLabel = ({
  appearance,
  help,
  helpValues,
  extra,
  inputId = '',
  label: inputLabel,
  labelValues,
  screenReaderOnly = false,
}: Props) => {
  const { formatMessage } = useIntl();

  const helpText =
    typeof help === 'object' ? formatMessage(help, helpValues) : help;
  const labelText =
    typeof inputLabel === 'object'
      ? formatMessage(inputLabel, labelValues)
      : inputLabel;
  return (
    <label
      className={getMainClasses(appearance, styles, {
        screenReaderOnly,
      })}
      id={inputId ? `${inputId}-label` : undefined}
      htmlFor={inputId || undefined}
    >
      <div>
        <span className={styles.labelText}>{labelText}</span>
        {helpText && <span className={styles.help}>{helpText}</span>}
      </div>
      {extra && <span>{extra}</span>}
    </label>
  );
};

InputLabel.displayName = displayName;

export default InputLabel;
