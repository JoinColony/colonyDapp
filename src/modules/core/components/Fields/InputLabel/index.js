/* @flow */

export { default } from './InputLabel';
