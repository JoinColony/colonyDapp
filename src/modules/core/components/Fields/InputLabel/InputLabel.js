/* @flow */

import { injectIntl } from 'react-intl';
import compose from 'recompose/compose';

import InputLabel from './InputLabel.jsx';

export default compose(injectIntl)(InputLabel);
