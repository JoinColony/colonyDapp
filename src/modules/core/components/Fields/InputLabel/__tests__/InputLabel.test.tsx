/* eslint-env jest */

import React from 'react';
import toJson from 'enzyme-to-json';

import { shallowWithIntl } from 'enzyme-react-intl';

import InputLabel from '../InputLabel';

describe('InputLabel intl={intl} component', () => {
  test('Renders initial component', () => {
    const wrapper = shallowWithIntl(
      <InputLabel appearance={{}} inputId="foo" help="halp" label="awesome" />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('If error is false, and help is true, returns help field', () => {
    const wrapper = shallowWithIntl(
      <InputLabel inputId="foo" help="(halp)" label="awesome" />,
    );
    expect(wrapper.html()).toContain('(halp)');
  });
});
