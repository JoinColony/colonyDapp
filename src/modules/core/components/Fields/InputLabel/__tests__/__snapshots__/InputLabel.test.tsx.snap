// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel intl={intl} component Renders initial component 1`] = `
<InputLabel
  appearance={Object {}}
  help="halp"
  inputId="foo"
  label="awesome"
>
  <label
    className=""
    htmlFor="foo"
    id="foo-label"
  >
    <span>
      awesome
    </span>
    <span>
      halp
    </span>
  </label>
</InputLabel>
`;
